ru-intl/remap/en/background.js

```diff
@@ -0,0 +1,199 @@
+                          "a": "â", "A": "Â", "i": "î", "I": "Î", "w": "ŵ", "W": "Ŵ", " ": "^",
+                          "z": "ẑ", "Z": "Ẑ", "s": "ŝ", "S": "Ŝ", "g": "ĝ", "G": "Ĝ", "c": "ĉ", "C": "Ĉ", },
+    "dead_breve"      : { "u": "ŭ", "U": "Ŭ", "o": "ŏ", "O": "Ŏ", "e": "ĕ", "E": "Ĕ", "a": "ă", "A": "Ă",
+                          "g": "ğ", "G": "Ğ", "i": "ĭ", "I": "Ĭ", " ": "˘", },
+    "dead_caron"      : { "u": "ǔ", "U": "Ǔ", "o": "ǒ", "O": "Ǒ", "e": "ě", "E": "Ě", "a": "ǎ", "A": "Ǎ",
+                          "n": "ň", "N": "Ň", "i": "ǐ", "I": "Ǐ", " ": "ˇ", "z": "ž", "Z": "Ž", "r": "ř",
+                          "R": "Ř", "s": "š", "S": "Š", "g": "ǧ", "G": "Ǧ", "k": "ǩ", "K": "Ǩ", "l": "ľ",
+                          "L": "Ľ", "c": "č", "C": "Č", },    
+    "dead_doubleacute": { "u": "ű", "U": "Ű", "o": "ő", "O": "Ő", " ": "˝", },
+    "dead_ogonek"     : { "u": "ų", "U": "Ų", "o": "ǫ", "O": "Ǫ", "e": "ę", "E": "Ę", "a": "ą", "A": "Ą",
+                          "i": "į", "I": "Į", " ": "˛", },
+    "dead_macron"     : { "u": "ū", "U": "Ū", "y": "ȳ", "Y": "Ȳ", "o": "ō", "O": "Ō", "e": "ē", "E": "Ē",
+                          "a": "ā", "A": "Ā", "i": "ī", "I": "Ī", " ": "¯", },
+    "dead_abovedot"   : { "y": "ẏ", "Y": "Ẏ", "o": "ȯ", "O": "Ȯ", "e": "ė", "E": "Ė", "a": "ȧ", "A": "Ȧ",
+                          "n": "ṅ", "N": "Ṅ", "i": "ı", "I": "İ", "w": "ẇ", "W": "Ẇ", " ": "˙",
+                          "z": "ż", "Z": "Ż", "r": "ṙ", "R": "Ṙ", "m": "ṁ", "M": "Ṁ", "p": "ṗ", "P": "Ṗ",
+                          "s": "ṡ", "S": "Ṡ", "g": "ġ", "G": "Ġ", "c": "ċ", "C": "Ċ", },
+    "dead_abovering"  : { "u": "ů", "U": "Ů", "y": "ẙ", "a": "å", "A": "Å", "w": "ẘ", " ": "°", },
+    "dead_cedilla"    : { "c": "ç", "C": "Ç", "s": "ş", "S": "Ş", "t": "ţ", "T": "Ţ", " ": "¸", },
+};
+
+var altstate = false;
+var deadkey = null;
+
+chrome.input.ime.onFocus.addListener(
+    function(context) {
+      contextID = context.contextID;
+    }
+);
+
+chrome.input.ime.onBlur.addListener(() => {
+  contextID = 0;
+})
+
+chrome.input.ime.onKeyEvent.addListener(
+    function(engineID, keyData) {
+      var handled = false;
+
+      if ((keyData.type == "keydown") && (keyData.code == "AltRight")) {
+          if (altstate == false) {
+              altstate = true;
+          }
+      }
+
+      if (keyData.type == "keydown" && keyData.altKey == false && keyData.ctrlKey == false) {
+
+          if (lut[keyData.code]) {
+            let modified = keyData.shiftKey ^ keyData.capsLock + 2 * altstate;
+            if (keys.includes(keyData.code)) {
+              modified = keyData.shiftKey + 2 * altstate;
+            }
+              
+            let emit = lut[keyData.code][modified];
+            altstate = false;
+            
+            if (deadkey != null) {
+                if (deadkeys[deadkey][emit]) {
+                  emit = deadkeys[deadkey][emit];
+                  deadkey = null;
```

Comment:
move this out of the if since both sides are the same?

Reply:
Done